common/pulp/common/error_codes.py

```diff
@@ -122,6 +122,9 @@
 PLP0042 = Error("PLP0042", _("This request is forbidden."), [])
 PLP0043 = Error("PLP0043", _("Database 'write_concern' config can only be 'majority' or 'all'. "
                              "Refer to /etc/pulp/server.conf for proper use."), [])
+PLP0044 = Error("PLP0044", _("Content unit type '%(content_unit)s' explicitly defines an index for"
+                             "its unit key. This is not allowed because the platform handles it for"
+                             "you."), ['content_unit'])
```

Comment:
A good reason not to put developer errors in the "coded exceptions" is that the coded exceptions need to be well documented and communicated to the users. As these users include our API consumers who may wish to respond to our error code, this will cause them to spend time thinking about how to handle a problem they will never experience. Let's save them the time by just raising a ValueError. Also, it's probably not necessary to use gettext since working on our code already implies that developers know English enough to read/write our docblocks and comments and variable/function names ☺